python/lsst/sconsUtils/state.py

```diff
@@ -98,8 +98,8 @@ def _initVariables():
         SCons.Script.BoolVariable('force', 'Set to force possibly dangerous behaviours', False),
         ('optfile', 'Specify a file to read default options from', None),
         ('prefix', 'Specify the install destination', None),
-        SCons.Script.EnumVariable('opt', 'Set the optimisation level', 0,
-                                  allowed_values=('0', '1', '2', '3')),
+        SCons.Script.EnumVariable('opt', 'Set the optimisation level', 3,
+                                  allowed_values=('g', '0', '1', '2', '3')),
```

Comment:
Does beg the question of why this particular set of allowed values but it's easy to add more if we need them. I don't think `Og` works on clang. clang seems to recommend `-O0 -g` but `-g` can be combined with all of them.